Better handling of file extensions in the documenten-api-uploader FormIO component (#238)
valtimo-platform/valtimo: documentation/features/zgw/zgw-documents/upload-to-documenten-api-with-metadata.md

The second service task in the Upload Document process is called 'Link uploaded document to zaak'. The Zaken API plugin can take care of this action. Look [here](../zgw-plugins/configure-zaken-api-plugin.md) for more information.

![Plugin action: Link uploaded document to zaak](../../../using-valtimo/upload/img/link-uploaded-document-to-zaak.png)

## Using the Documenten API File Upload FormIO component in user tasks

In the form builder you will find the Documenten API File Upload component under the advanced section.

> Drag and drop the component into your form to use it.

Reviewer: ```suggestion
To use it, simply drag and drop the component into your form and configure the properties as described below.
```

The following properties can be configured:
### Documenten API File Upload dropzone
![Documenten API File Upload Component](../../../.gitbook/assets/documenten-api-file-upload.png)
This is the dropzone that will be shown to the user in a user task, which can be customized by configuring the following properties:

> - **Label**: The label of the component. ***(required)***

Reviewer: ```suggestion
- **Label**: The label displayed for the component. ***(required)***
```

- **Required**: Whether the component is required.
- **Property Name**: The path to save the reference to the uploaded document in the case data. ***(required)***
- **Title**: The title of the component, shown in the dropzone of the uploader.
- **Hide Title**: Whether to hide the title of the component.
- **Subtitle**: The subtitle of the component, shown below the title.
- **Maximum file size**: The maximum file size in MB. ***(required)***
- **Hide maximum file size**: Whether to hide the maximum file size.
- **Allow camera uploads**: Whether to allow uploads from a camera connected to the device.
- **Disabled**: Whether the component is disabled.
- **Accepted file types**: The accepted file types in a comma-separated string.
- **Hide accepted file types**: Whether to hide the accepted file types.
- **Document URL**: The process variable name for saving the URL of the uploaded document.
### Metadata modal
![Documenten API Metadata Modal](../../../.gitbook/assets/documenten-api-metadata-modal.png)

> This is the modal that will be shown to the user when uploading a document, which can be customized by configuring the following properties:

Reviewer: ```suggestion
When a user uploads a document, a metadata modal is displayed.
This modal allows additional document metadata which can be customized by configuring the following properties:
```
